### DISCUSSION TITLE
Using yolov5 to detect objects in high-resolution (size= 1024 X 20000)images

### DISCUSSION THREAD

**Discussion by bishtdeepakhydro** [ultralytics/ultralytics#1806] [Q&A]
I have trained the yolo on custom object detection using normal 480x480 images. Can I use this model to detect objects in 1024x20000 resolution images. When directly passing image using "model(img)", it's not detecting the object.

PS. Do tell once the yolo is trained with weights in exp100 folder in " .pt" format, how can I save the model so that I can use it in offline mode again.

**Comment by glenn-jocher** [comment]
@bishtdeepakhydro any format can be used online or offline, there are no connectivity requirements.

You can run inference on any image size you want, but you'll get the best results when the objects in the predict images are the same size as the objects in the training images. Actual image sizes do not matter much, just the size of the objects inside them!

**Comment by bishtdeepakhydro** [comment]
Thank you for apt response.Kindly throw some light if the size of object is small than on images what is being trained on model, even if the shape or geometry is  similar. I was working on a custom model built from scratch including custom training of specific datasets. But eventually end up on yolov5 as it yielded better results in terms of detection, after going through other' models. This model was utilised for detecting anomalies in amplitude (wreck image) in water-fall image generated from acoustic reflections of Side Scan Sonar equipment. It would be appreciated if you could recommend any other suggestions on this.Regards. On 12-Apr-2023 5:53 am, Glenn Jocher ***@***.***> wrote:
@bishtdeepakhydro any format can be used online or offline, there are no connectivity requirements.
You can run inference on any image size you want, but you'll get the best results when the objects in the predict images are the same size as the objects in the training images. Actual image sizes do not matter much, just the size of the objects inside them!

**Reply by glenn-jocher** [reply]
@bishtdeepakhydro it's great to hear that YOLOv5 has been serving you well for your object detection tasks. When it comes to detecting smaller objects than those the model was trained on, there are a few considerations to keep in mind.

Firstly, if the objects in the high-resolution images are significantly smaller than the ones in the training set, the model might struggle to detect them due to the scale variance. In such cases, you might want to retrain your model with a more representative dataset that includes smaller objects similar in size to those in your high-resolution images.

Additionally, you could experiment with different model architectures or hyperparameters that are more sensitive to smaller objects. For instance, adjusting the anchor box sizes or the detection thresholds could potentially improve the detection of smaller objects.

Since you're working with specialized acoustic reflection data, you might also consider customizing the model architecture or employing domain-specific preprocessing techniques to enhance the model's ability to recognize the unique patterns in your data.

Remember, the key is to ensure that the training data closely matches the conditions of the data you wish to predict on, including object sizes and image resolutions. This will help the model generalize better to your specific use case.

If you're looking for further guidance on model optimization for small object detection or other advanced techniques, our documentation at https://docs.ultralytics.com provides a wealth of information that could be beneficial for your project. Good luck with your anomaly detection in acoustic reflections, and feel free to reach out if you have more questions! 🚀
